lib/Models/Catalog/CatalogItems/GeoJsonCatalogItem.ts

```diff
@@ -46,6 +51,45 @@ class GeoJsonCatalogItem
     return isDefined(this._file);
   }
 
+  /**
+   * Tries to build a FeatureCollection from partial geojson data
+   */
+  private featureCollectionFromPartialData(
+    jsonData: JsonValue
+  ): FeatureCollectionWithCrs | undefined {
```

Comment:
Extracted out previous code that builds a full `FeatureCollection` from possibly partial API responses as we have to now use it in a couple of places.